src/libvirt-xml-parse.js

```diff
@@ -560,7 +561,25 @@ function parseDumpxmlForSpice(devicesElem) {
 }
 
 function parseDumpxmlForTPM(devicesElem) {
-    return devicesElem.getElementsByTagName('tpm').length > 0;
+    const tpmElems = devicesElem.getElementsByTagName('tpm');
+    if (tpmElems.length > 0) {
+        const tpmElem = tpmElems[0];
+        const model = tpmElem.getAttribute('model');
+        const backendElem = tpmElem.getElementsByTagName('backend')[0];
+        const type = backendElem.getAttribute('type');
+        const version = backendElem.getAttribute('version');
+        const devicePathElem = backendElem.getElementsByTagName('device')[0];
+        const devicePath = devicePathElem ? devicePathElem.getAttribute('path') : null;
+
+        const tpm = {
+            model,
+            type,
+            version,
+            devicePath,
+        };
+        return { tpm, hasTPM: true };
+    }
+    return { tpm: {}, hasTPM: false };
```

Comment:
With just returning the `tpm` object here or `null`, this will be fully compatible with a boolean check. This commit *breaks* this (and is generally harder to use), hence my proposal to keep the current behaviour of being able to interpret the `parseDumpxmlForTPM()` result as a boolean.

Reply:
I understand. Thank you. I have made the corrections, so please check them.